### PR TITLE
add cover mask algorithm

### PR DESCRIPTION
**Pull request by zzzwjZhang**
Ref  #22759

Changes proposed in this pull request:
  - add KEEP_FIRST_N_LAST_M mask algorithm
  - add KEEP_FROM_X_TO_Y mask algorithm

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public void testNotSetStartIndex() {
+        KeepFirstNLastMMaskAlgorithm keepFirstNLastMMaskAlgorithm1 = new KeepFirstNLastMMaskAlgorithm();
+        Properties wrongProperties = new Properties();
+        wrongProperties.setProperty("stop-index", "5");
+        wrongProperties.setProperty("replace-char", "*");
+        assertThrows(IllegalArgumentException.class, () -> {
+            keepFirstNLastMMaskAlgorithm1.init(wrongProperties);
+        });
+    }
+
```
Comment (strongduanmu) [inline features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java:72]: Please remove this useless blank line.

File: features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    @Test
+    public void testMask() {
+        String result = keepFirstNLastMMaskAlgorithm.mask("abc123456");
+        assertEquals("ab*23456", result);
+    }
+    
+    @Test
+    public void testMaskIfPlainValueIsLess() {
+        String result = keepFirstNLastMMaskAlgorithm.mask("abc");
+        assertEquals("abc", result);
```
Comment (strongduanmu) [inline features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java:59]: Hi @zzzwjZhang, you can read ShardingSphere code conduct first - https://shardingsphere.apache.org/community/en/involved/conduct/code/. We recommend using assertThat over assertEquals.

Comment (zzzwjZhang) [reply features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java:59]: sorry,i will be careful next time.

Comment (strongduanmu) [reply features/mask/core/src/test/java/org/apache/shardingsphere/mask/algorithm/KeepFirstNLastMMaskAlgorithmTest.java:59]: Don't worry, I'm sure you'll do a great job.

**Review by strongduanmu — APPROVED** [review]
@zzzwjZhang Good job, merged.

**Pull request merged**